### PR TITLE
Handlers edit

### PR DESCRIPTION
**Pull request by arcogabbo**
#### List of Changes

Added more tests
Added more validations for headers

#### Motivation and Context

Improvements

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/__tests__/app.test.ts [hunk 1]
```diff
@@ -20,32 +23,41 @@ describe("app", () => {
...
   describe("GET /address-book-io/v1/digital-address/courtesy", () => {
     it("should return a 400 error if the x-pagopa-cx-taxid header is not provided", async () => {
-      const res = await app.get(endpoint);
+      const res = await app.get(endpoint).set({ "x-api-key": "key" });
       expect(res.status).toBe(400);
     })
 
     it("should return a 200 if the x-pagopa-cx-taxid header is provided", async () => {
-      const res = await app.get(endpoint).set({ "x-pagopa-cx-taxid": "example" });
-      expect(UserActivationDocument).toHaveBeenCalledTimes(1);
```
Comment (BurnedMarshal) [inline src/__tests__/app.test.ts:29]: Can we check that this method is called?

File: src/__tests__/app.test.ts [hunk 2]
```diff
@@ -20,32 +23,41 @@ describe("app", () => {
...
+      const res = await app.get(endpoint).set({ "x-pagopa-cx-taxid": aFiscalCode, "x-api-key": "key" });
       expect(res.status).toBe(200);
     })
+
+    it("should return a 403 error if the x-api-key header is not provided", async () => {
+      const res = await app.get(endpoint);
+      expect(res.status).toBe(403);
```
Comment (BurnedMarshal) [inline src/__tests__/app.test.ts:37]: Can we add this?
```suggestion
      expect(UserActivationDocument).not.toBeCalled();
      expect(res.status).toBe(403);
```
Here and in other failure tests where the User activation should not be called.

File: src/utils/__mocks__/functions_services_client.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import * as E from "fp-ts/lib/Either";
+
+const mockUpsert = jest.fn().mockImplementation(() => {
+  console.log("mock2")
```
[the remark this answers is not in the document]
Comment (arcogabbo) [reply src/utils/__mocks__/functions_services_client.ts]: done in https://github.com/pagopa/io-pn-mock/pull/2/commits/4eb86e5f0fa2c2d13fe42cc57a20254ca4631e84

File: src/__tests__/app.test.ts (hunk 2 above)
Comment (arcogabbo) [reply src/__tests__/app.test.ts:40]: done in https://github.com/pagopa/io-pn-mock/pull/2/commits/4eb86e5f0fa2c2d13fe42cc57a20254ca4631e84

File: src/__tests__/app.test.ts (hunk 1 above)
Comment (arcogabbo) [reply src/__tests__/app.test.ts:29]: done in https://github.com/pagopa/io-pn-mock/pull/2/commits/4eb86e5f0fa2c2d13fe42cc57a20254ca4631e84

File: package.json [hunk 4]
```diff
@@ -15,7 +15,8 @@
     "watch": "nodemon -r dotenv/config src/index.ts",
     "pregenerate": "bundle-api-spec --api-spec ./docs/openapi/pn-address-book-b2b-external-io-v1.yaml --out-path ./docs/openapi/bundled-openapi.yaml",
     "generate": "npm-run-all generate:*",
-    "generate:openapi": "gen-api-models --api-spec ./docs/openapi/bundled-openapi.yaml --out-dir ./src/generated/ --strict false"
+    "generate:openapi": "gen-api-models --api-spec ./docs/openapi/bundled-openapi.yaml --out-dir ./src/generated/ --strict false",
+    "generate:functions-services": "gen-api-models --api-spec https://raw.githubusercontent.com/pagopa/io-functions-services/master/openapi/index.yaml --out-dir ./src/generated --strict false --request-types --response-decoders --client"
```
[the remark this answers is not in the document]
Comment (arcogabbo) [reply package.json]: done in https://github.com/pagopa/io-pn-mock/pull/2/commits/4eb86e5f0fa2c2d13fe42cc57a20254ca4631e84

**Pull request merged**